modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaHttpServerCodegen.java

```diff
@@ -0,0 +1,486 @@
+    @SuppressWarnings("unchecked")
+    public static void marshallingPostProcessor(Map<String, Object> objs) {
+
+        if (objs == null) {
+            return;
+        }
+
+        Set<Marshaller> entityUnmarshallerTypes = new HashSet<>();
+        Set<Marshaller> entityMarshallerTypes = new HashSet<>();
+        Set<Marshaller> stringUnmarshallerTypes = new HashSet<>();
+        boolean hasCookieParams = false;
+        boolean hasMultipart = false;
+
+        Map<String, Object> operations = (Map<String, Object>) objs.get("operations");
+        if (operations != null) {
+            List<CodegenOperation> operationList = (List<CodegenOperation>) operations.get("operation");
+
+            for (CodegenOperation op : operationList) {
+                boolean isMultipart = op.isMultipart;
+                hasMultipart |= isMultipart;
+                hasCookieParams |= op.getHasCookieParams();
+                ArrayList<CodegenParameter> fileParams = new ArrayList<>();
+                ArrayList<CodegenParameter> nonFileParams = new ArrayList<>();
+                for (CodegenParameter parameter : op.allParams) {
+                    if (parameter.isBodyParam || parameter.isFormParam) {
+                        if (parameter.isFile) {
+                            fileParams.add(parameter.copy());
+                        } else {
+                            nonFileParams.add(parameter.copy());
+                        }
+                        if (!parameter.isPrimitiveType) {
+                            if (isMultipart) {
+                                stringUnmarshallerTypes.add(new Marshaller(parameter));
+                            } else {
+                                entityUnmarshallerTypes.add(new Marshaller(parameter));
+                            }
+                        }
+                    }
+                }
+                for (int i = 0, size = fileParams.size(); i < size; ++i) {
+                    fileParams.get(i).hasMore = i < size - 1;
+                }
+                for (int i = 0, size = nonFileParams.size(); i < size; ++i) {
+                    nonFileParams.get(i).hasMore = i < size - 1;
+                }
+
+                HashSet<Marshaller> operationSpecificMarshallers = new HashSet<>();
+                for (CodegenResponse response : op.responses) {
+                    if (!response.primitiveType) {
+                        Marshaller marshaller = new Marshaller(response);
+                        entityMarshallerTypes.add(marshaller);
+                        operationSpecificMarshallers.add(marshaller);
+                    }
+                    response.vendorExtensions.put("isDefault", response.code.equals("0"));
+                }
+                op.vendorExtensions.put("specificMarshallers", operationSpecificMarshallers);
+                op.vendorExtensions.put("fileParams", fileParams);
+                op.vendorExtensions.put("nonFileParams", nonFileParams);
```

Comment:
We're trying to normalize the naming of vendor extensions. I wonder if you can rename the extensions as followings:
- `specificMarshallers ` to `x-specific-marshallers `
- `fileParams` to `x-file-params`
- `nonFileParams` to `x-non-file-params`